runtime_scan/pkg/provider/docker/client.go

```diff
@@ -204,11 +207,17 @@ func (c *Client) createScanAssetVolume(ctx context.Context, config *provider.Sca
 	}()
 
 	// Export asset data to tar reader
-	assetContents, err := c.exportAsset(ctx, config)
+	assetContents, exportCleanup, err := c.exportAsset(ctx, config)
 	if err != nil {
 		return "", fmt.Errorf("failed to export asset: %w", err)
 	}
-	defer assetContents.Close()
+	defer func() {
+		err := assetContents.Close()
+		if err != nil {
+			logger.Errorf("failed to close asset contents stream: %s", err.Error())
+		}
+	}()
+	defer exportCleanup()
```

Comment:
No need to have two defers here. Also, there are valid returns with nil cleanup actions.
```go
defer func() {
	err := assetContents.Close()
	if err != nil {
		logger.Errorf("failed to close asset contents stream: %s", err.Error())
	}

	if exportCleanup != nil {
		exportCleanup()
	}
}()
```

Reply:
indeed! thanks, Ramiz!

Reply:
pushed 👍